### PR TITLE
[macOS SDK] Make changes necessary for Native Auth to be used from macOS

### PR DESCRIPTION
**Pull request by spetrescu84**
## Proposed changes

Made necessary changes for Native Auth to be used from macOS

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: IdentityCore/tests/automation/ui_tests_lib/XCUIElement+CrossPlat.m [hunk 1]
```diff
@@ -39,8 +39,8 @@ - (void)msidPasteText:(NSString *)text application:(__unused XCUIApplication *)a
 #if TARGET_OS_IPHONE
     [self typeText:text];
 #else
-    [[NSPasteboard generalPasteboard] declareTypes:[NSArray arrayWithObject:NSStringPboardType] owner:nil];
-    [[NSPasteboard generalPasteboard] setString:text forType:NSStringPboardType];
+    [[NSPasteboard generalPasteboard] declareTypes:[NSArray arrayWithObject:NSPasteboardTypeString] owner:nil];
```
Comment (nilo-ms) [inline IdentityCore/tests/automation/ui_tests_lib/XCUIElement+CrossPlat.m:42]: Why are those changes required? A more detailed PR description would be beneficial

Comment (spetrescu84) [reply IdentityCore/tests/automation/ui_tests_lib/XCUIElement+CrossPlat.m:42]: In macOS 10.15 the `NSStringPboardType` enum value has been replaced with `NSPasteboardTypeString`

File: IdentityCore/tests/util/MSIDTestCacheDataSource.h [hunk 2]
```diff
@@ -23,7 +23,7 @@
 
 #import <Foundation/Foundation.h>
 #import "MSIDExtendedTokenCacheDataSource.h"
-#import "MSIDBaseToken.h"
+#import "MSIDCredentialType.h"
```
Comment (spetrescu84) [inline IdentityCore/tests/util/MSIDTestCacheDataSource.h:26]: This change is needed because `MSIDBaseToken.h` cannot be properly imported by the Modulemap as it creates a circular reference

File: IdentityCore/tests/util/MSIDTestCacheDataSource.h [hunk 3]
```diff
@@ -23,7 +23,7 @@
...
 #import "MSIDCacheItemSerializing.h"
```
Comment (diegojerezba) [inline IdentityCore/tests/util/MSIDTestCacheDataSource.h:28]: [The CI is failing](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=1336850&view=logs&j=d46b93ad-1be7-5eab-6980-c78d65372179&t=0153d404-5636-5846-4ebd-361eb75124ec) here: `IdentityCore/src/workplacejoin/MSAIMSIDWorkPlaceJoinUtilBase`.

Does it mean we need to update [WorkplaceJoin](https://github.com/AzureAD/WorkplaceJoin-for-iOS) to 10.15 too?

File: IdentityCore/tests/automation/ui_tests_lib/XCUIElement+CrossPlat.m (hunk 1 above)
Comment (spetrescu84) [reply IdentityCore/tests/automation/ui_tests_lib/XCUIElement+CrossPlat.m]: I've moved them to the 10.15 PR because that's where they actually belong

File: IdentityCore/tests/util/MSIDTestCacheDataSource.h (hunk 3 above)
Comment (spetrescu84) [reply IdentityCore/tests/util/MSIDTestCacheDataSource.h:28]: Now it's working, not sure what caused it to fail. Apparently it is updated to 10.15

**Review by nilo-ms — APPROVED** [review]

**Review by diegojerezba — APPROVED** [review]

**Review by borgesmb — APPROVED** [review]

**Review by Veena11 — APPROVED** [review]
